### PR TITLE
Issue/edit

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: src/components/Todo/Todo.tsx [hunk 1]
```diff
@@ -18,6 +24,11 @@ export const Todo: React.FC<TodoProps> = ({ id, title, descr, deleteTodo }) => {
           btnClass="round-btn round-btn--red"
           onClickBtn={() => deleteTodo(id)}
         />
+        <RoundButton
+          btnText="E"
+          btnClass="round-btn round-btn--yellow"
+          onClickBtn={() => handleEditVisibleModal(id)}
```
Comment (igorTT) [inline src/components/Todo/Todo.tsx:30]: Does this function edit a visible modal?

**Review by igorTT — CHANGES_REQUESTED** [review]
Try to use modal type
```javascript
<ModalContainer
  ...
  type={modalType}
/>
```

File: src/components/Todo/Todo.tsx (hunk 1 above)
Comment (BelohvostovAlex) [reply src/components/Todo/Todo.tsx]: changed

**Review by igorTT — CHANGES_REQUESTED** [review]
When the user clicks 'edit' it should see the values that will be edited.
Only the field that was edited should be rewritten.

File: src/pages/TodoPage/TodoPageContainer.tsx [hunk 2]
```diff
@@ -5,8 +5,10 @@ import { ITodo } from '../../models/ITodo';
...
 export const TodoPageContainer: React.FC = () => {
   const [todos, setTodos] = useState([] as ITodo[]);
+  const [modalType, setModalType] = useState('');
   const [visibleModal, setVisibleModal] = useState(false);
-  const [visibleEditModal, setVisibleEditModal] = useState(false);
+  const [visibleModalWithEditFeature, setVisibleEditModalWithEditFeature] =
+    useState(false);
   const [currIDForEdit, setCurrIDForEdit] = useState('');
 
   const hasTodo = todos.length ? true : false;
```
[the remark this answers is not in the document]
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPageContainer.tsx:14]: changed according to the note

File: src/pages/TodoPage/TodoPageContainer.tsx [hunk 3]
```diff
@@ -21,15 +23,17 @@ export const TodoPageContainer: React.FC = () => {
...
+    setVisibleEditModalWithEditFeature(true);
     setCurrIDForEdit(id);
+    setModalType('edit');
   };
 
-  const onCloseEditVisibleModal = () => {
-    setVisibleEditModal(false);
+  const onCloseModalWithEditFeature = () => {
+    setVisibleEditModalWithEditFeature(false);
     setCurrIDForEdit('');
```
[the remark this answers is not in the document]
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPageContainer.tsx:37]: deleted

File: src/pages/TodoPage/TodoPageContainer.tsx [hunk 4]
```diff
@@ -5,9 +5,17 @@ import { ITodo } from '../../models/ITodo';
 
 export const TodoPageContainer: React.FC = () => {
   const [todos, setTodos] = useState([] as ITodo[]);
+  const [modalType, setModalType] = useState('');
   const [visibleModal, setVisibleModal] = useState(false);
+  const [visibleModalWithEditFeature, setVisibleEditModalWithEditFeature] =
```
Comment (igorTT) [inline src/pages/TodoPage/TodoPageContainer.tsx:10]: What's the difference between visibleModal and visibleModalWithEditFeature?

**Review by igorTT — CHANGES_REQUESTED** [review]
image
image

Please compare these two screenshots. They look pretty differently. Check the styles.

File: src/pages/TodoPage/TodoPageContainer.tsx (hunk 4 above)
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPageContainer.tsx:10]: emmm, literally the same logic, deleted

File: src/pages/TodoPage/TodoPageContainer.tsx [hunk 5]
```diff
@@ -19,6 +25,44 @@ export const TodoPageContainer: React.FC = () => {
...
+    setInitialValueForModalWithEditFeature({
+      id: currTodo!.id,
+      title: currTodo!.title,
+      description: currTodo!.description,
+    });
+  };
+
+  const onCloseModalWithEditFeature = () => {
+    setVisibleModal(false);
+  };
```
Comment (igorTT) [inline src/pages/TodoPage/TodoPageContainer.tsx:45]: Useless method. Replace it with 'handleVisibleModal' and nothing will be changed.
Create a handleCloseModal method and use it for closing modal. 
Also, try to pass on a method to the modal window that will be triggered on 'submit' action. Long story short. Modal window should get three parameters: onClose, onSubmit, initialValues

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/pages/TodoPage/TodoPageContainer.tsx (hunk 5 above)
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPageContainer.tsx:45]: tried to resolve the issue

File: src/components/Modal/ModalContainer.tsx [hunk 6]
```diff
@@ -8,23 +8,41 @@ import { Modal } from './Modal';
...
+    initialValueForModalWithEditFeature;
 
+  const [title, handleTitle] = useInput(
+    modalType === 'edit' ? initialTitle : ''
+  );
+  const [description, handleDescription] = useInput(
+    modalType === 'edit' ? initialDescription : ''
+  );
   const createTodo = () => {
     addTodo({ id: v4(), title: title, description: description });
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:27]: Pass only the title and description. Create id (in normal case- the id will be generated by db) in the handler.

File: src/components/Modal/ModalContainer.tsx [hunk 7]
```diff
@@ -8,23 +8,41 @@ import { Modal } from './Modal';
...
 
     onClose();
   };
+
+  const onSubmit = () => {
+    if (modalType === 'create') createTodo();
+    else editTodo(currIDForEdit, title, description);
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:34]: I don't see the reason for using this id at all. Try to delete it.
Don't use this kind of short expression. Use curly braces.

File: src/components/Modal/Modal.tsx [hunk 8]
```diff
@@ -46,12 +50,13 @@ export const Modal: React.FC<ModalProps> = ({
...
           <Button
-            text="Create"
+            text={type === 'create' ? 'Create' : 'Edit'}
+            type={ButtonTypes.SUBMIT}
             className="button button--blue"
-            onClick={createTodo}
           />
           <Button
             text="Cancel"
+            type={ButtonTypes.BUTTON}
```
Comment (igorTT) [inline src/components/Modal/Modal.tsx:59]: Add a default type value for the button.

File: src/components/Modal/ModalContainer.tsx [hunk 9]
```diff
@@ -8,23 +8,41 @@ import { Modal } from './Modal';
...
+  editTodo,
+  currIDForEdit,
+  modalType,
+  initialValueForModalWithEditFeature,
 }) => {
-  const [title, handleTitle] = useInput();
-  const [description, handleDescription] = useInput();
+  const { title: initialTitle } = initialValueForModalWithEditFeature;
+  const { description: initialDescription } =
+    initialValueForModalWithEditFeature;
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:18]: Destruct them in one expression.

File: src/components/Modal/ModalContainer.tsx [hunk 10]
```diff
@@ -8,23 +8,41 @@ import { Modal } from './Modal';
...
 
+  const [title, handleTitle] = useInput(
+    modalType === 'edit' ? initialTitle : ''
+  );
+  const [description, handleDescription] = useInput(
+    modalType === 'edit' ? initialDescription : ''
+  );
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:25]: When you invoke the creating a new todo- you don't add any initial value, You can use initial values without additional checks. Don't forget to erase the initial value, while closing the window

File: src/components/Modal/interfaces.tsx [hunk 11]
```diff
@@ -11,10 +11,18 @@ export interface ModalProps {
...
+  type: string;
 }
 
 export interface ModalContainerProps {
   onClose: () => void;
   addTodo: (todo: ITodo) => void;
+  editTodo: (id: string, title: string, description: string) => void;
+  currIDForEdit: string;
+  modalType: string;
+  initialValueForModalWithEditFeature: {
```
Comment (igorTT) [inline src/components/Modal/interfaces.tsx:24]: -> initialValue

File: src/components/Modal/interfaces.tsx [hunk 12]
```diff
@@ -11,10 +11,18 @@ export interface ModalProps {
...
-  createTodo: () => void;
+  onSubmit: () => void;
+  type: string;
 }
 
 export interface ModalContainerProps {
   onClose: () => void;
   addTodo: (todo: ITodo) => void;
+  editTodo: (id: string, title: string, description: string) => void;
+  currIDForEdit: string;
```
Comment (igorTT) [inline src/components/Modal/interfaces.tsx:22]: useless

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.tsx (hunk 8 above)
Comment (BelohvostovAlex) [reply src/components/Modal/Modal.tsx:59]: done

File: src/components/Modal/ModalContainer.tsx (hunk 9 above)
Comment (BelohvostovAlex) [reply src/components/Modal/ModalContainer.tsx:18]: done

File: src/components/Modal/ModalContainer.tsx (hunk 10 above)
Comment (BelohvostovAlex) [reply src/components/Modal/ModalContainer.tsx:25]: done, true story

File: src/components/Modal/ModalContainer.tsx (hunk 7 above)
Comment (BelohvostovAlex) [reply src/components/Modal/ModalContainer.tsx:34]: deleted, used curly braces

File: src/components/Modal/interfaces.tsx (hunk 11 above)
Comment (BelohvostovAlex) [reply src/components/Modal/interfaces.tsx:24]: changed

File: src/components/Modal/interfaces.tsx (hunk 12 above)
Comment (BelohvostovAlex) [reply src/components/Modal/interfaces.tsx:22]: changed

File: src/components/Button/Button.tsx [hunk 13]
```diff
@@ -10,7 +10,7 @@ export const Button: React.FC<ButtonProps> = ({
   onClick,
 }) => {
   return (
-    <button className={className} type={type} onClick={onClick}>
+    <button className={className} type={'button' || type} onClick={onClick}>
```
Comment (igorTT) [inline src/components/Button/Button.tsx:13]: Replace with 
```javascript
export const Button: React.FC<ButtonProps> = ({
  text,
  className,
  type = 'button',
  onClick,
}) => {
```

File: src/components/Modal/ModalContainer.tsx [hunk 14]
```diff
@@ -9,29 +8,22 @@ export const ModalContainer: React.FC<ModalContainerProps> = ({
...
 
     onClose();
   };
 
-  const onSubmit = () => {
-    if (modalType === 'create') createTodo();
-    else editTodo(currIDForEdit, title, description);
+  const onSubmitModalWithEditFeature = () => {
+    editTodo(title, description);
   };
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:27]: Replace with
```javascript
const handleSubmit = () => onSubmit({ title, description })
```

File: src/components/Modal/ModalContainer.tsx [hunk 15]
```diff
@@ -41,7 +33,9 @@ export const ModalContainer: React.FC<ModalContainerProps> = ({
       description={description}
       handleDescription={handleDescription}
       onClose={onClose}
-      onSubmit={onSubmit}
+      onSubmit={
+        modalType === 'create' ? createTodo : onSubmitModalWithEditFeature
+      }
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:38]: ->
```javascript
onSubmit={handleSubmit}
```

File: src/components/RoundButton/RoundButton.tsx [hunk 16]
```diff
@@ -10,7 +10,7 @@ export const RoundButton: React.FC<RoundButtonProps> = ({
   onClick,
 }) => {
   return (
-    <button className={className} onClick={onClick} type={type}>
+    <button className={className} onClick={onClick} type={'button' || type}>
```
Comment (igorTT) [inline src/components/RoundButton/RoundButton.tsx:13]: The same as above

**Review by igorTT — CHANGES_REQUESTED** [review]
You can move handleTitle and handleDescription to the modal.

File: src/components/Button/Button.tsx (hunk 13 above)
Comment (BelohvostovAlex) [reply src/components/Button/Button.tsx:13]: done

File: src/components/RoundButton/RoundButton.tsx (hunk 16 above)
Comment (BelohvostovAlex) [reply src/components/RoundButton/RoundButton.tsx:13]: done

**Comment by igorTT** [comment]
Check changes. If everything will be clear- merge pr.

**Pull request merged**